chain/epoch-manager/src/lib.rs

```diff
@@ -920,29 +926,34 @@ impl EpochManager {
         })
     }
 
-    /// Returns the list of chunk validators for the given shard_id and height.
-    pub fn get_chunk_validators(
+    /// Returns the list of chunk validators for the given shard_id and height and set of account ids.
+    /// Generation of chunk validators is deterministic and depends on the shard_id and height.
+    /// We cache the generated chunk validators and the corresponding set of account_ids.
+    pub fn get_chunk_validators_and_assignments(
         &self,
         epoch_id: &EpochId,
         shard_id: ShardId,
         height: BlockHeight,
-    ) -> Result<Arc<ChunkValidators>, EpochError> {
+    ) -> Result<(Arc<ChunkValidatorAssignments>, Arc<HashSet<AccountId>>), EpochError> {
```

Comment:
What do you think about including `HashSet<AccountId>` into `ChunkValidatorAssignments` as well? It will make it more obvious that these two structures represent the same set.
I'm also not absolutely sure that HashSet optimisation is necessary. But it looks better to be on the safe side.

Reply:
That's a great idea! I initially started with ChunkValidatorAssignments just to make a type alias but yeah, it can be a full struct! :)